### PR TITLE
[Java worker][Cpp worker]Support Java call Cpp Task

### PR DESCRIPTION
**Pull request by WangTaoTheTonic**
## Why are these changes needed?
Support calling cpp task from java side.

todo: doc (will write them down with all java/cpp xlang functions)

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by WangTaoTheTonic** [comment]
Failed test `//python/ray/dashboard:modules/serve/tests/test_serve_head ` is not related.

File: java/runtime/src/main/java/io/ray/runtime/AbstractRayRuntime.java [hunk 1]
```diff
@@ -178,6 +179,15 @@ public ObjectRef call(PyFunction pyFunction, Object[] args, CallOptions options)
         functionDescriptor, args, /*returnType=*/ Optional.of(pyFunction.returnType), options);
   }
 
+  @Override
+  public ObjectRef call(CppFunction cppFunction, Object[] args, CallOptions options) {
+    CppFunctionDescriptor functionDescriptor =
+        new CppFunctionDescriptor(cppFunction.functionName, "JAVA", "");
+    // Python functions always have a return value, even if it's `None`.
```
Comment (raulchen) [inline java/runtime/src/main/java/io/ray/runtime/AbstractRayRuntime.java:186]: This comment is copy-pasted. You should change it according to the C++ case. 
But I think the Python one is already out-dated. You can remove both of them.

File: java/test/src/main/java/io/ray/test/CrossLanguageInvocationTest.java [hunk 2]
```diff
@@ -170,6 +172,23 @@ public void testCallingPythonActor() {
     Assert.assertEquals(res.get(), "2".getBytes());
   }
 
+  @Test
+  public void testCallingCppFunction() {
+    ObjectRef<Integer> res = Ray.task(CppFunction.of("Plus", Integer.class), 1, 2).remote();
```
Comment (raulchen) [inline java/test/src/main/java/io/ray/test/CrossLanguageInvocationTest.java:177]: Can you add some comments? E.g., the first case is "Test calling a simple C++ function.", the second is "Test calling a C++ function that returns a large object."

**Review by raulchen — APPROVED** [review]
LGTM. a few small comments.

File: java/runtime/src/main/java/io/ray/runtime/AbstractRayRuntime.java (hunk 1 above)
Comment (WangTaoTheTonic) [reply java/runtime/src/main/java/io/ray/runtime/AbstractRayRuntime.java]: ok

**Pull request merged**